pkg/kubeadm/master/addons.go

```diff
@@ -101,6 +105,9 @@ func createKubeDNSPodSpec(s *kubeadmapi.KubeadmConfig) api.PodSpec {
 	)
 
 	return api.PodSpec{
+		NodeSelector: map[string]string{
+			"beta.kubernetes.io/arch": runtime.GOARCH,
```

Comment:
dns is tied to the master, so locking dns to the current architecture of kubeadm is safe